### ISSUE TITLE
Add `{{mode}}` Template Variable

### ISSUE THREAD

**Issue by joealden** [wxt-dev/wxt#562]
### Feature Request

Config options like [`zip.artifactTemplate`](https://wxt.dev/api/wxt/interfaces/InlineConfig.html#zip) allow template variable usage such as `{{name}}`, `{{version}}` etc.

It'd be great if a `{{mode}}` variable could be added that mirrors the Vite mode passed.

#### Is your feature request related to a bug?

N/A

### What are the alternatives?

N/A

### Additional context

N/A

**Comment by aklinker1** [comment]
Good idea! This would be a great first issue for anyone contributing.

Here's the function that takes in a string, and applies the template variables:

https://github.com/wxt-dev/wxt/blob/05db264e02bcf78ac50746454670a7c3344e3d3e/src/core/zip.ts#L33-L41

We'll also need to update the JSDocs here:

https://github.com/wxt-dev/wxt/blob/05db264e02bcf78ac50746454670a7c3344e3d3e/src/types/index.ts#L115-L140

And that should be it... There don't appear to be unit tests around the `applyTemplate` function, and it's simple enough I don't think it's necessary. But an E2E test could be good to make sure we're passing in the right variables and actually consuming the template. It would go in `e2e/tests/zip.test.ts`.

To run the test:

```sh
pnpm test e2e/tests/zip.test.ts
```

But if someone opens a PR, I can help add an E2E test.

And if you want to run it manually to verify everything is working, update the `demo/wxt.config.ts` file, and run:

```sh
cd demo
pnpm zip -b firefox
```

**Comment by GuiEpi** [comment]
I have created a PR for this request #566

**Comment by aklinker1** [comment]
Released in [v0.17.9](https://github.com/wxt-dev/wxt/releases/tag/v0.17.9)
